src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSource.cs

```diff
@@ -546,14 +545,7 @@ public void Dispose()
                 }
                 finally
                 {
-                    lock (_throttleLock)
-                    {
-                        if (_throttle != null)
-                        {
-                            _throttle.Release();
-                            _throttle = null;
-                        }
-                    }
+                    Interlocked.Exchange(ref _throttle, null)?.Release();
```

Comment:
The _throttle may be shared globally, but this private ThrottledResponse wrapper object doesn't seem to be shared. It can call _throttle.Release() without worrying about another thread modifying _throttle. So I still don't think that Interlocked.Exchange should be used.